src/pybind/mgr/cephadm/services/iscsi.py

```diff
@@ -138,3 +139,14 @@ def ok_to_stop(self,
         names = [f'{self.TYPE}.{d_id}' for d_id in daemon_ids]
         warn_message = f'It is presumed safe to stop {names}'
         return HandleCommandResult(0, warn_message, '')
+
+    def purge(self, service_name: str) -> None:
+        """Removes the gateway.conf from the iscsi pool
+        """
+        spec = cast(IscsiServiceSpec, self.mgr.spec_store[service_name].spec)
+        try:
+            with self.mgr.rados.open_ioctx(spec.pool) as ioctx:
+                ioctx.remove_object("gateway.conf")
+                logger.debug(f'<gateway.conf> removed from {spec.pool}')
+        except rados.ObjectNotFound:
```

Comment:
I think we should also allow other exceptions here. Like access forbidden or something. I don't see a point in retrying calling `purge` till it succeeds. 

```suggestion
        except rados.Errror:
            logger.exception('failed to purge {service_name}')
```